lib/core/device.rb

```diff
@@ -853,8 +853,14 @@ def switch_window(action)
   #   Value
   def switch_frame(action)
     index = action["Value"]
-    log_info("#{@role}: Switching to frame: #{index}")
-    if index == "parent"
+    el = wait_for(action)
```

Comment:
Is `wait_for` handled okay if we only specify Type and Value? Also I think there should be extra handling to prevent using Strategy/Id in combination with Value.

Reply:
Yes, you are right, I have corrected it, and I have added tests to cover this functionality